### ISSUE TITLE
gethostbyname error on macos 10.15 (github actions vm)

### ISSUE THREAD

**Issue by jakebolewski** [JuliaParallel/MPI.jl#407]
We are seeing a github action macos 10.15 issue where calling MPI.Init() fails:

```
Fatal error in MPI_Init: Invalid group, error stack:
MPIR_Init_thread(586)..............: 
MPID_Init(224).....................: channel initialization failed
MPIDI_CH3_Init(105)................: 
MPID_nem_init(324).................: 
MPID_nem_tcp_init(175).............: 
MPID_nem_tcp_get_business_card(401): 
MPID_nem_tcp_init(373).............: gethostbyname failed, Mac-1594849612293 (errno 0)
(unknown)(): Invalid group
SingleStackUtils: Error During Test at /Users/runner/work/ClimateMachine.jl/ClimateMachine.jl/test/testhelpers.jl:16
  Test threw exception
  Expression: mpiexec() do cmd
    run(`$cmd $oversubscribe -n $ntasks $(Base.julia_cmd()) --startup-file=no --project=$(Base.active_project()) $file`)
    true
end
  failed process: Process(`/Users/runner/.julia/artifacts/848ee2ddce903941ae946cd49f63eac561bd636d/bin/mpiexec -n 1 /Users/runner/hostedtoolcache/julia/1.4.2/x64/bin/julia -Cnative -J/Users/runner/work/ClimateMachine.jl/ClimateMachine.jl/ClimateMachine.so --check-bounds=yes -g1 --startup-file=no --project=/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/jl_pVc3V7/Project.toml /Users/runner/work/ClimateMachine.jl/ClimateMachine.jl/test/Utilities/SingleStackUtils/ssu_tests.jl`, ProcessExited(8)) [8]
```

https://github.com/CliMA/ClimateMachine.jl/runs/875221461#step:8:169

Reading the source code, it seems like defining MPICH_INTERFACE_HOSTNAME as localhost should override gethostbyname behavior, but unfortunately this doesn't seem to fix the issue.

`gethostbyname` is an older posix function that has since been replaced with `getaddrinfo` so this might be fixed in the upcoming 3.4 release candidate.

https://github.com/pmodels/mpich/issues/2889

**Comment by jakebolewski** [comment]
cc @simonbyrne

**Comment by simonbyrne** [comment]
If you're having this problem, it seems the workaround for now is to use the environment variables:
```
MPICH_INTERFACE_HOSTNAME: localhost
JULIA_MPIEXEC_ARGS: "-host localhost"
```

**Comment by jkozdon** [comment]
Just ran into this, is there some other fix or is this still the way to go?

**Comment by jkozdon** [comment]
(to be clear the fix worked... just wondering.)

**Comment by simonbyrne** [comment]
It should be fixed by the latest MPICH, I removed the workaround in #477

**Comment by fredrikekre** [comment]
I still see this on my local machine (macos 11.6.6) with MPICH_jll v4.0.2+2` but upstream says it should be fixed in 3.4?
